it/controllers/pensions/pensionIncome/StatePensionControllerISpec.scala

```diff
@@ -0,0 +1,419 @@
+          textOnPageCheck(user.specificExpectedResults.get.expectedDetailsBullet1, detailsBulletList(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedDetailsBullet2, detailsBulletList(2))
+          formPostLinkCheck(statePension(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+          errorSummaryCheck(user.specificExpectedResults.get.expectedErrorMessage, Selectors.yesSelector)
+          errorAboveElementCheck(user.specificExpectedResults.get.expectedErrorMessage, Some("value"))
+
+        }
+      }
+    }
+
+
+    "redirect to Pensions Summary page when user selects 'yes' and not a prior submission" which {
+      lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.yes)
+      lazy val result: WSResponse = {
+        dropPensionsDB()
+        authoriseAgentOrIndividual(isAgent = false)
+        val statePensionModel = anStateBenefitViewModelOne.copy(
+          amountPaidQuestion = None
+        )
+        val pensionViewModel = anIncomeFromPensionsViewModel.copy(statePension = Some(statePensionModel))
+        insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(incomeFromPensions = pensionViewModel)), aUserRequest)
+        urlPost(fullUrl(statePension(taxYearEOY)), body = form, follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "has a SEE_OTHER(303) status" in {
+        result.status shouldBe SEE_OTHER
+        //ToDo should redirect to Amount paid question
+        result.header("location") shouldBe Some(controllers.pensions.routes.PensionsSummaryController.show(taxYearEOY).url)
+      }
+
+      "updates amount paid question to Some(true)" in {
+        lazy val cyaModel = findCyaData(taxYearEOY, aUserRequest).get
+        cyaModel.pensions.incomeFromPensions.statePension.flatMap(_.amountPaidQuestion) shouldBe Some(true)
+      }
+    }
+  }
+
+  "redirect to Pension Summary when user selects 'no' and doesnt complete CYA model" which {
+    lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.no)
+    lazy val result: WSResponse = {
+      dropPensionsDB()
+      authoriseAgentOrIndividual(isAgent = false)
+      userDataStub(anIncomeTaxUserData, nino, taxYearEOY)
+
+      val statePensionModel = anStateBenefitViewModelOne.copy(
+        amountPaidQuestion = Some(true),
+        amount = Some(100.00)
+      )
+      val pensionViewModel = anIncomeFromPensionsViewModel.copy(statePension = Some(statePensionModel))
+      insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(incomeFromPensions = pensionViewModel)), aUserRequest)
+      urlPost(fullUrl(statePension(taxYearEOY)), body = form, follow = false,
+        headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+    }
+
+    "has a SEE_OTHER(303) status" in {
+      result.status shouldBe SEE_OTHER
+      result.header("location") shouldBe Some(pensionSummaryUrl(taxYearEOY))
```

Comment:
Seems you need to add a todo as well here - (should redirect to cya page when user selects No)